File: Problems/No.705_Design_HashSet/design_hashset.py
class MyHashSet(object):

    def __init__(self):
        self.HashSet = {}


    def add(self, key):
        """
        :type key: int
        :rtype: None
        """
        self.HashSet[key] = True
        

    def remove(self, key):
        """
        :type key: int
        :rtype: None
        """
        self.HashSet[key] = False
        

    def contains(self, key):
        """
        :type key: int
        :rtype: bool
        """
        if key in self.HashSet:
            return self.HashSet[key]
        else:
            return False
        


# Your MyHashSet object will be instantiated and called as such:
# obj = MyHashSet()
# obj.add(key)
# obj.remove(key)
# param_3 = obj.contains(key)

File: Problems/No.705_Design_HashSet/design_hashset.cpp
class MyHashSet {

private:
    std::unordered_map<int, bool> HashSet;

public:
    MyHashSet() {}
    
    void add(int key) {HashSet[key] = true;}
    
    void remove(int key) {HashSet[key] = false;}
    
    bool contains(int key) {

        if (HashSet.count(key)) return HashSet[key];
        else return false;

    }
};

/**
 * Your MyHashSet object will be instantiated and called as such:
 * MyHashSet* obj = new MyHashSet();
 * obj->add(key);
 * obj->remove(key);
 * bool param_3 = obj->contains(key);
 */
